src/controller/CHIPDeviceController.h

```diff
@@ -533,6 +537,15 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
        If no device is currently being paired, this value will be kNumMaxPairedDevices.  */
     uint16_t mDeviceBeingPaired;
 
+    enum CertificateChainType : uint16_t
```

Comment:
This, along with a lot of the other device attestation business logic and data marshalling procedures, should live in a separate code module, called by CHIPDeviceController. I do not think it belongs in DeviceController since it's purely a data model consideration of the specifics of the OperationalCredentials cluster.

Reply:
I think it should be part of the Controller flow. Besides, we've already moved some of the stuff to the crypto/ layer and (will move to) ConfigurationManager Class